### PR TITLE
Use native secp256k1 and altbn128 if present

### PR DESCRIPTION
**Pull request by shemnon**
Via JNA use the native scep256k1 and altbn128 libraries if it is present. 

Signed-off-by: Danno Ferrin <danno.ferrin@gmail.com>

### CODE REVIEW

File: crypto/src/main/java/org/hyperledger/besu/crypto/LibSecp256k1.java [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+  public static final int SECP256K1_CONTEXT_VERIFY = 0x0101;
+  public static final int SECP256K1_CONTEXT_SIGN = 0x0201;
+
+  /* Flag to pass to secp256k1_ec_pubkey_serialize. */
+  public static final int SECP256K1_EC_UNCOMPRESSED = 0x0002;
+
+  static final LibSecp256k1 INSTANCE = findInstance();
+  static final PointerByReference CONTEXT = createContext();
+
+  private static LibSecp256k1 findInstance() {
```
Comment (CjHare) [inline crypto/src/main/java/org/hyperledger/besu/crypto/LibSecp256k1.java:45]: Maybe an `info` or `debug` log statement in `findInstance()` would make sense, to let the user know that either native is or is not being used?

File: crypto/src/main/java/org/hyperledger/besu/crypto/SECP256K1.java [hunk 2]
```diff
@@ -308,6 +336,99 @@ public static Bytes calculateECIESKeyAgreement(final PrivateKey privKey, final P
...
+    final ByteBuffer recoveredKey = ByteBuffer.allocate(65);
+    final LongByReference keySize = new LongByReference(recoveredKey.limit());
+    LibSecp256k1.INSTANCE.secp256k1_ec_pubkey_serialize(
+        LibSecp256k1.CONTEXT,
+        recoveredKey,
+        keySize,
+        newPubKey,
+        LibSecp256k1.SECP256K1_EC_UNCOMPRESSED);
+
+    return Optional.of(PublicKey.create(Bytes.wrapByteBuffer(recoveredKey).slice(1)));
```
Comment (RatanRSur) [inline crypto/src/main/java/org/hyperledger/besu/crypto/SECP256K1.java:429]: Does the native `recoveredKey` return null in the same cases as the java implementation?

Comment (shemnon) [reply crypto/src/main/java/org/hyperledger/besu/crypto/SECP256K1.java:423]: Legacy APIs.  The old java method had a way to return emptys, but when I try and return empty from any of the methods that return error I get reference tests that break.  I'm not sure why the java method gets away with it.

**Comment by shemnon** [comment]
Also, please merge on approval if I'm asleep.

**Pull request merged**